### ISSUE TITLE
Warning when using onInputChange from FileUpload component with React 16

### ISSUE THREAD

**Issue by lnunno** [palantir/blueprint#1949]
### Bug report

Based off the docs for the [FileUpload component](http://blueprintjs.com/docs/#core/components/forms/file-upload.javascript-api) it looks like onInputChange is a property, however I get a warning when using it with React 16.

Packages:

- blueprintjs/core: "1.35.0",
- react: "16.2.0"

#### Steps to reproduce

```
<FileUpload
    text="Choose file to upload..."
    onInputChange={this.upload}
/>
```

#### Actual behavior

Seeing this warning on the console when using the FileUpload component:

```
Warning: Unknown event handler property `onInputChange`. It will be ignored.
    in label (created by Blueprint.FileUpload)
    in Blueprint.FileUpload (created by VanillaDataNavigator)
    in div (created by VanillaDataNavigator)
    in VanillaDataNavigator (created by Connect(VanillaDataNavigator))
    in Connect(VanillaDataNavigator) (created by MosaicRoot)
    in div (created by InternalMosaicWindow)
    in div (created by InternalMosaicWindow)
    in InternalMosaicWindow (created by DragSource(InternalMosaicWindow))
    in DragSource(InternalMosaicWindow) (created by DropTarget(DragSource(InternalMosaicWindow)))
    in DropTarget(DragSource(InternalMosaicWindow)) (created by ViewIdMosaicWindow)
    in ViewIdMosaicWindow (created by MosaicRoot)
    in div (created by MosaicRoot)
    in div (created by MosaicRoot)
    in MosaicRoot (created by Mosaic)
    in div (created by Mosaic)
    in Mosaic (created by DragDropContext(Mosaic))
    in DragDropContext(Mosaic) (created by App)
    in App
    in Provider
```

#### Expected behavior

No warning should be emitted.

**Comment by giladgray** [comment]
this is a simple bug in how we're constructing the `...htmlProps` object in `FileInput.tsx` (note component rename for 2.0), just gotta make sure to remove `onInputChange` too.

**Comment by lnunno** [comment]
@giladgray Is there a fix I can add to my code or is this an issue internal to blueprint?

**Comment by giladgray** [comment]
it's internal to blueprint but should be very easy to resolve if you want to submit a PR. just strip `onInputChange` from the `...htmlProps` spread, as mentioned above.

**Comment by giladgray** [comment]
fixed in #2023
